apps/transport/lib/jobs/multi_validation_with_error_notification_job.ex

```diff
@@ -26,34 +28,56 @@ defmodule Transport.Jobs.MultiValidationWithErrorNotificationJob do
     |> Enum.each(fn {%DB.Dataset{} = dataset, multi_validations} ->
       dataset
       |> emails_list()
+      |> MapSet.difference(notifications_sent_recently(dataset))
       |> Enum.each(fn email ->
```

Comment:
Si je comprends bien le `save_notification` est fait juste après l'envoi, je n'étais pas sûr au début.

Ca évite que si une erreur se produit en cours d'itération, qu'on re-spamme les gens déjà notifiés.

Il faut par contre espérer qu'on n'aura pas d'erreur ecto à la sauvegarde (sinon il faut un autre système avec une pré-sauvegarde juste avant l'envoi, et un marquage juste après, mais ça me paraît un peu overkill pour l'instant).

Reply:
C'est bien ça, c'est pour ça que je pense que c'est safe au retry